Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/TransformToolIntegrationTest.java

```diff
@@ -1191,4 +1191,48 @@ public void testRotateMultipleLayersUndoRedoWhenRotatingWasNotLastCommand() {
 		onDrawingSurfaceView()
 				.checkLayerDimensions(initialHeight, initialWidth);
 	}
+
+	@Test
+	public void testCropWithClickOutsideToolbox() {
+		onToolBarView()
+				.performSelectTool(ToolType.BRUSH);
+
+		onDrawingSurfaceView()
+				.perform(swipe(DrawingSurfaceLocationProvider.HALFWAY_TOP_LEFT, DrawingSurfaceLocationProvider.HALFWAY_BOTTOM_RIGHT));
+		onDrawingSurfaceView()
+				.perform(swipe(DrawingSurfaceLocationProvider.HALFWAY_TOP_RIGHT, DrawingSurfaceLocationProvider.HALFWAY_BOTTOM_LEFT));
+
+		onToolBarView()
+				.performSelectTool(ToolType.TRANSFORM);
+
+		onTransformToolOptionsView()
+				.performAutoCrop();
+
+		onDrawingSurfaceView()
+				.perform(touchAt(DrawingSurfaceLocationProvider.TOOL_POSITION));
+
+		onDrawingSurfaceView()
+				.perform(swipe(DrawingSurfaceLocationProvider.LEFT_MIDDLE,
+						DrawingSurfaceLocationProvider.HALFWAY_RIGHT_MIDDLE));
+		onDrawingSurfaceView()
+				.perform(touchAt(DrawingSurfaceLocationProvider.OUTSIDE_MIDDLE_LEFT));
+
+		onDrawingSurfaceView()
+				.perform(swipe(DrawingSurfaceLocationProvider.RIGHT_MIDDLE,
+						DrawingSurfaceLocationProvider.HALFWAY_RIGHT_MIDDLE));
+		onDrawingSurfaceView()
+				.perform(touchAt(DrawingSurfaceLocationProvider.OUTSIDE_MIDDLE_RIGHT));
+
+		onDrawingSurfaceView()
+				.perform(swipe(DrawingSurfaceLocationProvider.BOTTOM_MIDDLE,
+						DrawingSurfaceLocationProvider.HALFWAY_BOTTOM_MIDDLE));
+		onDrawingSurfaceView()
+				.perform(touchAt(DrawingSurfaceLocationProvider.OUTSIDE_MIDDLE_BOTTOM));
+
+		onDrawingSurfaceView()
+				.perform(swipe(DrawingSurfaceLocationProvider.TOP_MIDDLE,
+						DrawingSurfaceLocationProvider.HALFWAY_TOP_MIDDLE));
+		onDrawingSurfaceView()
+				.perform(touchAt(DrawingSurfaceLocationProvider.OUTSIDE_MIDDLE_TOP));
```

Comment:
You are not actually verifying any change to the tool. This test should fail without your changes to production, but right now I guess it will pass.